File: router_annotations/src/main/java/com/dede/router/annotations/Route.java
package com.dede.router.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by hsh on 2019-06-03 15:20
 * 路由定义注解
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.CLASS)
public @interface Route {

    String[] route();

    boolean parserParams() default true;

    boolean typeCase() default true;
}

File: router_annotations/src/main/java/com/dede/router/annotations/Route.kt
package com.dede.router.annotations

/**
 * Created by hsh on 2019-06-03 15:20
 * 路由定义注解
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.SOURCE)
annotation class Route(val route: Array<String>, val parserParams: Boolean = true, val typeCase: Boolean = true)
